src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/StrandArtifactFilter.java

```diff
@@ -43,30 +44,40 @@
     public ErrorType errorType() { return ErrorType.ARTIFACT; }
 
     @Override
-    public double calculateErrorProbability(final VariantContext vc, final Mutect2FilteringEngine filteringEngine, ReferenceContext referenceContext) {
-        final EStep probabilities = calculateArtifactProbabilities(vc, filteringEngine);
-        return probabilities.forwardArtifactResponsibility + probabilities.reverseArtifactResponsibility;
+    public List<Double> calculateErrorProbabilityForAlleles(final VariantContext vc, final Mutect2FilteringEngine filteringEngine, ReferenceContext referenceContext) {
+        final List<EStep> alleleProbs = calculateArtifactProbabilities(vc, filteringEngine);
+        return alleleProbs.isEmpty() ? Collections.emptyList() :
+                alleleProbs.stream().map(probabilities -> probabilities.forwardArtifactResponsibility + probabilities.reverseArtifactResponsibility).collect(Collectors.toList());
     }
 
-    public EStep calculateArtifactProbabilities(final VariantContext vc, final Mutect2FilteringEngine filteringEngine) {
-        // {fwd ref, rev ref, fwd alt, rev alt}
-        final int[] counts = filteringEngine.sumStrandCountsOverSamples(vc, true, false);
-
-        final int indelSize = Math.abs(vc.getReference().length() - vc.getAlternateAllele(0).length());
-        if (counts[2] + counts[3] == 0 || indelSize > LONGEST_STRAND_ARTIFACT_INDEL_SIZE) {
-            return new EStep(0, 0, counts[0] + counts[2], counts[1] + counts[3], counts[2], counts[3]);
+    public List<EStep> calculateArtifactProbabilities(final VariantContext vc, final Mutect2FilteringEngine filteringEngine) {
+        // for each allele, forward and reverse count
+        List<List<Integer>> sbs = StrandBiasUtils.getSBsForAlleles(vc);
+        if (sbs == null || sbs.isEmpty() || sbs.size() <= 1) {
+            return Collections.emptyList();
         }
 
-
-        return strandArtifactProbability(strandArtifactPrior, counts[0] + counts[2], counts[1] + counts[3], counts[2], counts[3], indelSize);
-
+        final ListIterator<Integer> indelSizeIterator = vc.getAlternateAlleles().stream().map(alt -> Math.abs(vc.getReference().length() - alt.length())).collect(Collectors.toList()).listIterator();
+        int totalFwd = sbs.stream().map(sb -> sb.get(0)).reduce(0, Math::addExact);
+        int totalRev = sbs.stream().map(sb -> sb.get(1)).reduce(0, Math::addExact);
+        // skip the reference
+        List<List<Integer>> altSBs = sbs.subList(1, sbs.size());
+
+        return altSBs.stream().map(altSB -> {
+            final int altIndelSize = indelSizeIterator.next();
```

Comment:
Incrementing an iterator as a side-effect of a lambda spooks me.  It feels like it's adhering to the letter of the law that requires variables inside a lambda to be final but violating the spirit of functional programming.  I would rather have an indel size list and have an `IntStream` even at the cost of an extra line `altSB = altSBs.get(n)`.